file/file.go

```diff
@@ -212,6 +216,62 @@ func NewWriter(pth string, opt *Options) (w Writer, err error) {
 	return w, err
 }
 
+type Iterator struct {
+	err     error
+	isValid bool
+	reader  Reader
+}
+
+func NewIterator(path string, opts *Options) *Iterator {
+	r, err := NewReader(path, opts)
+	return &Iterator{
+		err:     err,
+		isValid: err == nil,
+		reader:  r,
+	}
+}
+
+func (i *Iterator) Lines() iter.Seq[[]byte] {
+	// only iterator if the reader is properly set.
```

Comment:
shouldn't iterator be iterate?
```go
// only iterate if the reader is properly set.   
```

Reply:
iter.Seq[[]byte] is the iterator function. We could call this whatever makes the most sense
 1. `for l := range it.Lines()`
 2. `for b := range it.Iterate()`
 3. `for b := range it.Range()`

I wasn't sure if this should return a string or a []byte. I think []byte is more standard.